Cargo.toml

```diff
@@ -17,11 +17,15 @@ keywords = ["bgp", "bgpkit"]
 [dependencies]
 reqwest = {version = "0.11", default-features = false, features = ["blocking", "json"]}
 serde = {version="1.0", features = ["derive"]}
-oneio = {version="0.14", default-features = false, features=["json"]}
```

Comment:
👋 Hi @digizeph, I hope you don't mind me chiming in on a merged pull request. 

I'm debugging an issue where native-tls got pulled back into the dependency tree for my project and I _think_ it may be related to the removal of `default-features = false` here. I could be wrong, but I believe this is because of oneio's [default dependencies](https://github.com/bgpkit/oneio/blob/v0.15.2/Cargo.toml#L50-L52) pulling in native tls:

```rust
[features]
default = ["lib"]
lib = ["remote", "compressions", "json", "lib-native-tls"]
```

Would you be open to a PR that adds back `default-features = false`?

Reply:
Of course! Can you try the following patch to see if that works for you? On my side, this got rid of the native-tls dependency: `cargo tree --no-default-features --features rustls|grep native`

```
Index: Cargo.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	(revision Staged)
+++ b/Cargo.toml	(date 1701221807220)
@@ -18,8 +18,7 @@
 reqwest = {version = "0.11", default-features = false, features = ["blocking", "json"]}
 serde = {version="1.0", features = ["derive"]}
 serde_json = "1"
-
-oneio = "0.15.2"
+oneio = {version="0.15.2", default-features = false, features=["json", "remote", "compressions"]}
 regex = "1"
 anyhow = "1.0"
 chrono = "0.4"
```

Reply:
I went ahead cut a new release `v0.4.3` that should resolve this:
https://github.com/bgpkit/bgpkit-commons/releases/tag/v0.4.3

I also added the following checks to make sure
* `native-tls` does not bring `rustls` around: https://github.com/bgpkit/bgpkit-commons/blob/472e80ba96107041308a2edab2e613f18da665a7/.github/workflows/rust.yaml#L27
* `rustls` does not bring `native-tls` around: https://github.com/bgpkit/bgpkit-commons/blob/472e80ba96107041308a2edab2e613f18da665a7/.github/workflows/rust.yaml#L30

Reply:
That did the trick, thank you @digizeph!